tests/browser/test/connecting_blocks_test.js

```diff
@@ -0,0 +1,112 @@
+
+let browser;
+suite("Testing Connecting Blocks", function(done) {
+  this.timeout(0);
+  // Setup Selenium for each of the test
+  suiteSetup(async function() {
+    const options = {
+      capabilities: {
+        "browserName": 'chrome',
+        'goog:chromeOptions': {
+          args: ['--allow-file-access-from-files'],
+        },
+      },
+      services: [
+        ['selenium-standalone'],
+      ],
+      logLevel: 'warn',
+    };
+  
+    // Run in headless mode on Github Actions.
+    if (process.env.CI) {
+      options.capabilities['goog:chromeOptions'].args.push(
+          '--headless', '--no-sandbox', '--disable-dev-shm-usage',);
+    } else {
+      // --disable-gpu is needed to prevent Chrome from hanging on Linux with
+      // NVIDIA drivers older than v295.20. See
+      // https://github.com/google/blockly/issues/5345 for details.
+      options.capabilities['goog:chromeOptions'].args.push('--disable-gpu');
+    }
+  // Use Selenium to bring up the page
+  const url = 'file://' + posixPath(path.join(__dirname, '..')) + '/index.html';
+    console.log('Starting webdriverio...');
+    browser = await webdriverio.remote(options);
+    console.log('Loading URL: ' + url);
+    await browser.url(url);
+    return browser;
+  });
+
+test('Testing Block Connection', async function() {
+    await new Promise((resolve) => setTimeout(resolve, 2000)); // 2 sec
+    let flyOutBlock = await browser.$('#blocklyDiv > div > svg:nth-child(6) > g > g.blocklyBlockCanvas > g:nth-child(13)');
+    await flyOutBlock.dragAndDrop({x: 300, y: 0});
+
+    let flyOutBlock2 = await browser.$('#blocklyDiv > div > svg:nth-child(6) > g > g.blocklyBlockCanvas > g:nth-child(15)');
+    await flyOutBlock2.dragAndDrop({x: 300, y: 0});
+
+    
+    // Wait for the DOM to settle and get the flyout blocks on the Blockly Canvas
+    await new Promise((resolve) => setTimeout(resolve, 2000)); // 2 sec
+
+     flyOutBlock = await browser.$('#blocklyDiv > div > svg.blocklySvg > g > g.blocklyBlockCanvas > g:nth-child(3)');
+     flyOutBlock2 = await browser.$('#blocklyDiv > div > svg.blocklySvg > g > g.blocklyBlockCanvas > g:nth-child(4)');
+ 
+
+     await flyOutBlock.dragAndDrop(flyOutBlock2);
+     // The accepeted radius from origin is 28 so moving between that and the edge of the block
+     await flyOutBlock.dragAndDrop({x: 35, y: 0, duration: 2000});
+
+     await new Promise((resolve) => setTimeout(resolve, 2000)); // 2 sec
+
+    // Make sure the block that was appended now has a block that is connected
+    const blocksConnectedCheck = await browser.$('#blocklyDiv > div > svg.blocklySvg > g > g.blocklyBlockCanvas > g:nth-child(3) > g.blocklyDraggable');
+
+    const blocksConnected = await browser.execute(() => {
+      const parentBlock = Blockly.getMainWorkspace().getTopBlocks(false)[1];
+      if (childBlock.getChildren()) {
+        return true;
+      } else {
+        return false;
+      }
+   });
+
+    chai.assert.isTrue(blocksConnected);
```

Comment:
This is very cool! I like asserting based on the workspace state rather than the dom state a lot better.

When we finalize this, my preference is that this is the approach we go with!